src/integration-test/java/com/commercetools/sync/integration/services/impl/CartDiscountServiceImplIT.java

```diff
@@ -0,0 +1,425 @@
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class CartDiscountServiceImplIT {
+
+    private CartDiscountService cartDiscountService;
+
+    private List<String> errorCallBackMessages;
+    private List<Throwable> errorCallBackExceptions;
+
+    /**
+     * Cleans up the target test data that were built in this test class.
+     */
+    @AfterAll
+    static void tearDown() {
+        deleteTypes(CTP_TARGET_CLIENT);
+        deleteCartDiscountsFromTargetAndSource();
+    }
+
+    /**
+     * Deletes types & cart discounts from the target CTP project, then it populates the project with test data.
+     */
+    @BeforeEach
+    void setup() {
+        deleteTypes(CTP_TARGET_CLIENT);
+        deleteCartDiscountsFromTargetAndSource();
+
+        populateTargetProject();
+
+        errorCallBackMessages = new ArrayList<>();
+        errorCallBackExceptions = new ArrayList<>();
+
+        final CartDiscountSyncOptions cartDiscountSyncOptions = CartDiscountSyncOptionsBuilder
+                .of(CTP_TARGET_CLIENT)
+                .errorCallback((errorMessage, exception) -> {
+                    errorCallBackMessages.add(errorMessage);
+                    errorCallBackExceptions.add(exception);
+                })
+                .build();
+
+        cartDiscountService = new CartDiscountServiceImpl(cartDiscountSyncOptions);
+    }
+
+    @Test
+    void fetchCartDiscount_WithExistingCartDiscountKey_ShouldFetchCartDiscount() {
+        final Optional<CartDiscount> cartDiscountOptional = CTP_TARGET_CLIENT
+                .execute(CartDiscountQuery.of()
+                        .withPredicates(cartDiscountQueryModel ->
+                                cartDiscountQueryModel.name().lang(Locale.ENGLISH).is(CART_DISCOUNT_KEY_1)))
+                .toCompletableFuture().join().head();
+        assertThat(cartDiscountOptional).isNotNull();
+
+        final Optional<CartDiscount> fetchedCartDiscountOptional =
+                cartDiscountService.fetchCartDiscount(CART_DISCOUNT_KEY_1).toCompletableFuture().join();
+
+        assertThat(fetchedCartDiscountOptional).isEqualTo(cartDiscountOptional);
+    }
+
+    @Test
+    void fetchCartDiscount_WithBlankKey_ShouldNotFetchCartDiscount() {
+        final Optional<CartDiscount> fetchedCartDiscountOptional =
+                cartDiscountService.fetchCartDiscount("").toCompletableFuture().join();
+        assertThat(fetchedCartDiscountOptional).isEmpty();
+    }
+
+    @Test
+    void fetchCartDiscount_WithNullKey_ShouldNotFetchCartDiscount() {
+        final Optional<CartDiscount> fetchedCartDiscountOptional =
+                cartDiscountService.fetchCartDiscount(null).toCompletableFuture().join();
+        assertThat(fetchedCartDiscountOptional).isEmpty();
+    }
+
+
+    @Test
+    void fetchMatchingCartDiscountsByKeys_WithEmptySetOfKeys_ShouldReturnEmptySet() {
```

Comment:
Do you need IT for this case? You already have unit test in the other PR. This IT doesn't add any value. Or you can remove unit tests. WDYT?

Reply:
this is not in unit test just fetchCartDiscount methods.

Reply:
I agree a unit test for this would be better instead of an IT. Because then we won't need to call the expensive `@BeforeEach` here